text/0019-beforeSendTransaction.md

```diff
@@ -0,0 +1,102 @@
+
+- Lack of ultimate control over transaction events
+  - Event processors happen in an unspecified order, and there's no guarantee a processor added by a user will run last. This means that an integration, for example, might change event data after the user's last chance to intervene.
+
+(That last point is not _strictly_ true. As a last resort, a user could create a custom transport or could proxy events through their own servers, but both of those are pretty ugly, cumbersome workarounds.)
+
+# Background
+
+When transactions were introduced, the decision was made not to run them through `beforeSend` because they follow a slightly different schema, and therefore had the potential to break any existing `beforeSend` which relies on its input being a certain shape. It's unclear whether a transaction-specific `beforeSend`-type hook was discussed at the time.
+
+# Supporting Data
+
+Issues where this has come up:
+- https://github.com/getsentry/sentry-docs/issues/5525
+- https://github.com/getsentry/sentry-javascript/issues/4723
+- https://github.com/getsentry/sentry-javascript/issues/5442
+- https://github.com/getsentry/sentry-python/issues/1226
+
+(and I'm sure many others)
+
+# Options Considered
+
+_(Revised after preliminary discussion in https://github.com/getsentry/rfcs/pull/19)_
+
+### **Option 1**
+
+_(Original proposal, demoted to one option of several after discussion)_
+
+Add `beforeSendTransaction` to all SDKs, which (as the name implies) would work exactly the same way `beforeSend` does, but would act upon transactions.
+
+Pros:
+- Not a breaking change, so less pain for customers in the short run.
+- Not a breaking change, so no need to wait for a major release to implement it.
+
+Cons:
+- Increased API surface.
+- Code that acts on shared event properties (tags, contexts, etc) would need to be duplicated between `beforeSend` and `beforeSendTransaction`.
+- Unclear/non-parallel naming when considered alongside `beforeSend`. (`beforeSendTransaction` is a clear name, but `beforeSend` is not - it really ought to be `beforeSendError`; similar problem to current confusion with `sampleRate` and `tracesSampleRate`.)
+- Future payload types (replays, profiles, etc) would need their own methods. Where does it end?
+
+### **Option 2**
+
+In the next major of each SDK, start sending both errors and transactions through `beforeSend`.
+
+Pros:
+- `beforeSend` has a seemingly-universally-applicable name, so the most intuitive thing for users is to have it fulfill its implied role and apply universally.
+- Future payload types could  be added without introducing new init options/versions of the method.
+
+Cons:
+- Potentially breaking change for everyone, even people not using tracing, because old assumptions about `beforeSend` input schema would no longer hold. (We could maybe work around this by just surrounding our `beforeSend` call with a `try-catch` and returning the event untouched if things go sideways.)
```

Comment:
IMO another con is that users might end up checking the type of the payload and then casting it to different payload types.

Reply:
If they did this, would it break downstream processing in the SDK?

Reply:
I don't think it would, but IMO this is an unappealing API.